File: libs/language/src/lib/mapping/__tests__/schema/x1.ts
import { z } from 'zod';
import { xSchema } from './x';
import { withTypeBlueId } from '@blue-company/schema-annotations';

export const x1Schema = withTypeBlueId('X1-BlueId')(
  xSchema.extend({
    intArrayField: z.array(z.number()).optional(),
    stringListField: z.array(z.string()).optional(),
    integerSetField: z.set(z.number()).optional(),
  })
);

File: references/java-sdk/src/test/java/blue/language/mapping/model/X1.java
package blue.language.mapping.model;

import blue.language.model.TypeBlueId;

import java.util.List;
import java.util.Set;

@TypeBlueId("X1-BlueId")
public class X1 extends X {
    public int[] intArrayField;
    public List<String> stringListField;
    public Set<Integer> integerSetField;
}
